src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -174,6 +166,24 @@ IOptions<FeatureConfiguration> featureConfiguration
             {
                 _storeResponseBuilder.SetWarningMessage(DicomCoreResource.IndexedDicomTagHasMultipleValues);
             }
+
+            // If there is an error in the core tag, return immediately
+            if (storeValidatorResult.InvalidTagErrors.Any(x => x.Value.IsRequiredTag) ||
+                !_enableDropInvalidDicomJsonMetadata && storeValidatorResult.InvalidTagErrors.Any())
+            {
+                ushort failureCode = FailureReasonCodes.ValidationFailure;
+                LogValidationFailedDelegate(_logger, index, failureCode, null);
+                _storeResponseBuilder.AddFailure(dicomDataset, failureCode, storeValidatorResult);
+                return null;
+            }
+            else if (_enableDropInvalidDicomJsonMetadata)
```

Comment:
This can be a separate `if` statement since the previous one returns.

Reply:
Done